pkg/api/query/v1_test.go

```diff
@@ -72,8 +73,21 @@ type endpointTestCase struct {
 	response interface{}
 	errType  baseAPI.ErrorType
 }
+type responeCompareFunction func(interface{}, interface{}) bool
 
-func testEndpoint(t *testing.T, test endpointTestCase, name string) bool {
+// Does a deep compare for two http responses.
+func deepCompare(a interface{}, b interface{}) bool {
+	return reflect.DeepEqual(a, b)
+}
```

Comment:
Why do we need this function? reflect.DeepEqual has the same signature, right?

Reply:
Yeah it does, but I wanted to check if `Stats` is being returned in query result and this could'nt be achieved if we do a deepEqual as Stats gives non consistent values hence added a compare function with same signature and added `lookupStats` function

Reply:
Also any idea why e2e is failing ? :|

Reply:
> Why do we need this function? reflect.DeepEqual has the same signature, right?

Oh, I think I got your point. I'll fix it :p